src/include/daos_api.h

```diff
@@ -97,16 +98,17 @@ daos_tx_open(daos_handle_t coh, daos_handle_t *th, uint64_t flags,
  *			Possible error values include:
  *			-DER_NO_HDL     invalid transaction handle.
  *			-DER_INVAL      Invalid parameter
- *			-DER_RESTART	transaction conflict detected.
+ *			-DER_TX_RESTART	transaction needs to restart (e.g.,
+ *					due to conflicts).
  */
 int
 daos_tx_commit(daos_handle_t th, daos_event_t *ev);
 
 /**
  * Create a read-only transaction from a snapshot. This does not create the
- * snapshot, but only a read transaction to be able to read from a persistent
- * snapshot in the container. If the user passes an epoch that is not
- * snapshoted, or the snapshot was deleted, reads using that transaction might
+ * snapshot, but only a read transaction to be able to read from a snapshot
+ * created with daos_cont_create_snap. If the user passes an epoch that is not
+ * snapshoted, or the snapshot was deleted, reads using that transaction may
  * fail if the epoch was aggregated.
```

Comment:
would they fail?  I guess if it's a conditional read and the object no longer exists at the epoch in question it would fail.   Unless failure is defined as getting incorrect results?   It may be better to just say that reads using that epoch may get unexpected or undefined results.

Reply:
Agreed. Let me write a new patch to change that.